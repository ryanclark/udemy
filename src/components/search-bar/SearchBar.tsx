import React from 'react';
import styled from 'react-emotion';

interface SearchBarProps {
  onChange: (value: string) => void;
}

const Input = styled('input')`
  background: #fff;
  font-size: 32px;
  color: #545e6f;
  border: none;
  padding: 30px 40px;
  outline: none;
  width: 100%;
`;

export class SearchBar extends React.Component<SearchBarProps> {
  ref: HTMLInputElement;

  componentDidMount() {
    // auto focus input on mount
    this.ref.focus();
  }

  private handleChange(event: React.ChangeEvent<HTMLInputElement>) {
    const { value } = event.target;

    this.props.onChange(value);
  }

  render() {
    return (
      <Input
        innerRef={(ref) => this.ref = ref}
        placeholder="Search characters..."
        onChange={(event) => this.handleChange(event)}
      />
    );
  }
}
