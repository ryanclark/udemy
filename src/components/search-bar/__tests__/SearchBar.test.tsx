import React from 'react';

import { mount } from 'enzyme';

import { SearchBar } from '../SearchBar';

describe('Search Bar', () => {
  it('should render correctly', () => {
    const wrapper = mount(<SearchBar value="" onChange={() => null} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should trigger the handler when the input is changed', () => {
    const onChange = jest.fn();
    const wrapper = mount(<SearchBar value="" onChange={onChange} />);
    const value = 'Test text';

    wrapper.find('input').simulate('change', { target: { value } });

    expect(onChange).toBeCalledWith(value);
  });
});
