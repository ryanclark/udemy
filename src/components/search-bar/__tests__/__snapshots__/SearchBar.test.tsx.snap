// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar should render correctly 1`] = `
<SearchBar
  onChange={[Function]}
>
  <Styled(input)
    innerRef={[Function]}
    onChange={[Function]}
    placeholder="Search characters..."
  >
    <input
      className="css-1bkzmhc"
      onChange={[Function]}
      placeholder="Search characters..."
    />
  </Styled(input)>
</SearchBar>
`;
